pkg/provision/engine.go

```diff
@@ -121,6 +144,19 @@ func (e *Engine) Run(ctx context.Context) error {
 				log.Error().Err(err).Msg("failed to updated the capacity counters")
 			}
 
+			if reservation.last && !triggeredCleanupOnBoot {
+				// Trigger cleanup by sending a struct onto the channel
+				cleanUp <- struct{}{}
```

Comment:
I wonder if there is not a chance of dead lock here. If the `cleanUp` channel is already full, then this line will block and prevent the execution from reading the`cleanUp` channel to allow to continue here => dead lock